### ISSUE TITLE
Can anybody get the m38u link for this vod? It was deleted by the streamer earlier today

### ISSUE THREAD

**Issue by gameinn1** [TwitchRecover/TwitchRecover#368]
https://twitchtracker.com/icrazyji/streams/47383415901

This is the twithc tracker link. Thank you.

**Comment by ghost** [comment]
@gameinn1 

[VodRecovery_icrazyji_47383548701.txt](https://github.com/TwitchRecover/TwitchRecover/files/9894118/VodRecovery_icrazyji_47383548701.txt)

I can't attach an M3U8 file so you are going to have to change the file extension to .M3U8... after that the vod should be playable. 

The stream was split into 2 parts... 
https://streamscharts.com/channels/icrazyji/streams/47383415901 - 55 minutes (Not Available)

https://streamscharts.com/channels/icrazyji/streams/47383548701 - 6 hours 5 minutes (Available)
First 4 segments are not playable which is why I attached the text file above. After removing the first 4 segments the vod is 6 hours 1 minute long.

**Comment by gameinn1** [comment]
> @gameinn1
> 
> [VodRecovery_icrazyji_47383548701.txt](https://github.com/TwitchRecover/TwitchRecover/files/9894118/VodRecovery_icrazyji_47383548701.txt)
> 
> I can't attach an M3U8 file so you are going to have to change the file extension to .M3U8... after that the vod should be playable.
> 
> The stream was split into 2 parts... https://streamscharts.com/channels/icrazyji/streams/47383415901 - 55 minutes (Not Available)
> 
> https://streamscharts.com/channels/icrazyji/streams/47383548701 - 6 hours 5 minutes (Available) First 4 segments are not playable which is why I attached the text file above. After removing the first 4 segments the vod is 6 hours 1 minute long.

@ItIckeYd 
Thanks. Do you know why this program failed to give me a m38u link and briefly explain the process you went through to get this? Also the vod plays fine in vlc but is there anyway to download it?

**Comment by ghost** [comment]
@gameinn1 The stream was split up into 2 parts.. You tried the first one which didn't exist as I stated above.

In order to produce the correct result you need to check how many segments are valid then edit the m3u8 file and comment out any invalid segments.

Technically the only way to do this is programmatically (Create a script or program).

In this case there was a total 1776 segments of which 1772 were valid.

I had to alter the m3u8 file and comment out segments 0,1,2,3 in order for the vod to play properly. Which is why I attached the file instead of giving you a link... 

I'm not sure about downloading through files.... you may need to look that up. I believe VLC has functionality which can actually convert to mp4 however it takes a while.

**Comment by gameinn1** [comment]
@ItIckeYd 
Yes but how are you supposed to get the second url from twitch tracker to make the 5 option (recover a vod) work in the program? I kind of understand it via streamcharts cause it does show 2 separate vods but when I paste in 

https://streamscharts.com/channels/icrazyji/streams/47383548701 (the ~6 hour vod)

the program just instantly closes.

**Comment by ghost** [comment]
@gameinn1 I created my own script that I update/maintain. I referenced this in #340... https://github.com/ItIckeYd/VodRecovery
